include/thread.h

```diff
@@ -103,6 +103,8 @@ typedef struct thread {
   /* signal handling */
   sigset_t td_sigpend; /* Pending signals for this thread. */
   /* TODO: Signal mask, sigsuspend. */
+
+  bool td_uses_fpu;
```

Comment:
Please use `td_flags` instead - that's preferred way to store thread flags.

You should read surrounding code more carefully for your changes to integrate well.